pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1490,4 +1489,41 @@ self: super: {
   x509-validation = dontCheck super.x509-validation;
   tls = dontCheck super.tls;
 
+  # Bump requested on upstream https://github.com/bgamari/monoidal-containers/pull/62
+  monoidal-containers = overrideCabal super.monoidal-containers (drv: {
+    postPatch = (drv.postPatch or "") + ''
+      substituteInPlace monoidal-containers.cabal --replace "these >= 1 && <1.1" "these -any"
+    '';
+  });
+
+  # Fixing test and deps on upstream on https://github.com/reflex-frp/patch/pull/17
+  patch = dontCheck (doJailbreak (overrideCabal (super.patch.override {
+    dependent-map = super.dependent-map_0_3_1_0;
+  }) (drv: {
+    postPatch = (drv.postPatch or "") + ''
+      substituteInPlace patch.cabal --replace "these >=1 && <1.1" "these -any"
+    '';
+  })));
+
+  # Bumped bounds in upstream PR https://github.com/reflex-frp/reflex/pull/434
+  # Tests disabled because of very old dep: https://github.com/reflex-frp/reflex/issues/435
+  reflex = dontCheck (doJailbreak (overrideCabal (super.reflex.override {
+    dependent-map = super.dependent-map_0_3_1_0;
+  }) (drv: {
+    postPatch = (drv.postPatch or "") + ''
+      substituteInPlace reflex.cabal --replace "haskell-src-exts >=1.16 && <1.23" "haskell-src-exts -any"
+      substituteInPlace reflex.cabal --replace "these >=1 && <1.1" "these -any"
+      substituteInPlace reflex.cabal --replace "dependent-sum >=0.6 && <0.7" "dependent-sum -any"
+    '';
+  })));
```

Comment:
In general, I think @peti prefers these types of things to be done in a `fetchpatch`, so it is obvious when they are no longer needed (when the patch stops applying cleanly).

There should be a bunch of examples in this file of using `fetchpatch` to pull down patches from a given PR.

(Same for the `reflex-dom-core` thing below.)

Reply:
Is fetchpatch on an upstream PR in general preferable to e.g. jailbreaking?

Reply:
I have decided that yes.

Reply:
> Is fetchpatch on an upstream PR in general preferable to e.g. jailbreaking?

I think there are two different things going on here:

1. Is it preferable to use `fetchpatch` over `substituteInPlace` for fixing versioning problems?

    Peti has said that he prefers `fetchpatch` in these situations so we know exactly when the patch is no longer valid.

1. Is it preferable to use `fetchpatch` over `doJailbreak`?

    This is less straightforward, and more depends on the situation.  In general, `fetchpatch` is preferable, but there are lots of Haskell libraries that don't particularly follow the latest LTS (instead they follow an older LTS), so sending a PR to them is just annoying.  They don't particularly appreciate it, because they don't intend to chase the latest LTS.  In these cases, doJailbreak can be easier.